src/base/ActionsRouter.sol

```diff
@@ -0,0 +1,143 @@
+/// If additional actions are required, define new unique action constants, and implement the logic for handling them
+/// in the `_handleAdditionalActions{1/2}` functions
+/// The _msgSender function returns the msg.sender from the call to executeActions. This allows actions to know which
+/// address instructed the action to be requested. The msg.sender within the unlockCallback with be the v4 PoolManager.
+abstract contract BaseActionsRouter is SafeCallback, ReentrancyLock {
+    using TransientStateLibrary for IPoolManager;
+    using SafeTransferLib for address;
+
+    /// @notice emitted when different numbers of parameters and actions are provided
+    error LengthMismatch();
+
+    /// @notice emitted when an inheriting contract does not support an action
+    error UnsupportedAction(uint256 action);
+
+    constructor(IPoolManager poolManager) SafeCallback(poolManager) {}
+
+    /// @notice internal function that triggers the execution of a set of actions on v4
+    function _executeActions(bytes calldata params) internal isNotLocked {
+        poolManager.unlock(params);
+    }
+
+    /// @notice function that is called by the PoolManager through the SafeCallback.unlockCallback
+    function _unlockCallback(bytes calldata data) internal override returns (bytes memory) {
+        // TODO decode in calldata directly for gas
+        (uint256[] memory actions, bytes[] memory params) = abi.decode(data, (uint256[], bytes[]));
+
+        uint256 numActions = actions.length;
+        if (numActions != params.length) revert LengthMismatch();
+
+        for (uint256 actionIndex = 0; actionIndex < numActions; actionIndex++) {
+            uint256 action = actions[actionIndex];
+
+            // separate actions on a pool, and settling deltas
+            if (action < 0x10) {
+                if (action == Actions.SWAP) _swap(params[actionIndex]);
+                else if (action == Actions.INCREASE_LIQUIDITY) _increaseLiquidity(params[actionIndex]);
+                else if (action == Actions.DECREASE_LIQUIDITY) _decreaseLiquidity(params[actionIndex]);
+                else if (action == Actions.MINT_POSITION) _mintPosition(params[actionIndex]);
+                else if (action == Actions.BURN_POSITION) _burnPosition(params[actionIndex]);
+                else if (action == Actions.DONATE) _donate(params[actionIndex]);
+                else _handleAdditionalActions2(action, params[actionIndex]);
+            } else {
+                if (action == Actions.SETTLE) _settle(params[actionIndex]);
+                else if (action == Actions.TAKE) _take(params[actionIndex]);
+                else if (action == Actions.MINT_6909) _mint6909(params[actionIndex]);
+                else if (action == Actions.BURN_6909) _burn6909(params[actionIndex]);
+                else if (action == Actions.CLEAR_DELTA) _clearDelta(params[actionIndex]);
+                else if (action == Actions.SWEEP_ETH) _sweepETH(params[actionIndex]);
```

Comment:
sweepEth is irrelevant - should be handled not inside the unlockCallback..

These should only be actions relevant to calling a function on v4